pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -302,10 +313,27 @@ private void readMorePolicies(SystemTopicClient.Reader<PulsarEvent> reader) {
     private void refreshTopicPoliciesCache(Message<PulsarEvent> msg) {
         if (EventType.TOPIC_POLICY.equals(msg.getValue().getEventType())) {
             TopicPoliciesEvent event = msg.getValue().getTopicPoliciesEvent();
-            policiesCache.put(
-                    TopicName.get(event.getDomain(), event.getTenant(), event.getNamespace(), event.getTopic()),
-                    event.getPolicies()
-            );
+            TopicName topicName =
+                    TopicName.get(event.getDomain(), event.getTenant(), event.getNamespace(), event.getTopic());
+            switch (msg.getValue().getActionType()) {
+                case INSERT:
+                    TopicPolicies old = policiesCache.putIfAbsent(topicName, event.getPolicies());
+                    if (old != null) {
+                        log.warn("Policy insert failed, the topic: {}' policy already exist", topicName);
+                    }
+                    break;
+                case UPDATE:
+                    policiesCache.put(topicName, event.getPolicies());
+                    break;
```

Comment:
It looks like can be combined into one method

Reply:
Hi, cound you describe it detaily. 😄

Reply:
```
case INSERT:                    
case UPDATE:
    TopicPolicies old = policiesCache.putIfAbsent(topicName, event.getPolicies());
    if (old != null) {
        log.warn("Policy insert failed, the topic: {}' policy already exist", topicName);
    }
    break;
```

Reply:
Oh, my mistake, forget it